vnext/Microsoft.ReactNative/Fabric/Composition/TextInput/WindowsTextInputComponentView.cpp

```diff
@@ -1518,4 +1522,108 @@ winrt::Microsoft::ReactNative::ComponentView WindowsTextInputComponentView::Crea
   return winrt::make<WindowsTextInputComponentView>(compContext, tag, reactContext);
 }
 
+void WindowsTextInputComponentView::autoCapitalizeOnUpdateProps(const std::string &capitalizationType) noexcept {
+  /*
+    Possible values are:
+     Characters - All characters.
+     Words - First letter of each word.
+     Sentences - First letter of each sentence.
+     None - Do not autocapitalize anything.
+  */
+
+  BSTR bstr = nullptr;
+  winrt::hresult hr = m_textServices->TxGetText(&bstr);
+
+  if (SUCCEEDED(hr)) {
+    std::wstring wstrText = std::wstring(bstr, SysStringLen(bstr));
+
+    if (capitalizationType == "characters") {
+      std::transform(wstrText.begin(), wstrText.end(), wstrText.begin(), towupper);
+    } else if (capitalizationType == "words") {
+      bool nextWord = true;
+
+      for (auto &ch : wstrText) {
+        if (std::iswspace(ch)) {
+          nextWord = true;
+        } else if (nextWord) {
+          ch = std::towupper(ch);
+          nextWord = false;
+        }
+      }
+    } else if (capitalizationType == "sentences") { // DOUBLE CHECK THIS LOGIC
+      bool newSentence = true;
+
+      // Patterns to look at to declare newSentence as true: first character, or "? ", or "! ", or ". ".
```

Comment:
Does a new line count as a new Sentence? I would test this out on a multiline TextInput (mostly because I always miss that edgecase lol)!

Reply:
also does this work if you copy/paste text into the TextInput?

Reply:
Capitalization rules doesn't apply to characters inserted in the middle of the string, except if `autoCapitalize='characters'`.

Testing the multiline scenario is a good observation, let me check what iOS/Android do and add new TextInputs to the E2E test app.